src/plugins/Multipart.js

```diff
@@ -62,7 +62,12 @@ export default class Multipart extends Plugin {
           this.core.log(`Download ${file.name} from ${file.uploadURL}`)
           return resolve(file)
         } else {
-          this.core.emitter.emit('core:upload-error', file.id)
+          // OPTION 1: return the full XHR object
+          this.core.emitter.emit('core:upload-error', file.id, xhr);
+
+          // OPTION 2: return the payload of the XHR object
+          this.core.emitter.emit('core:upload-error', file.id, xhr.response);
```

Comment:
I think will be more flexible to attach the xhr object. Thus you have the payload of the request and the error code.